projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts

```diff
@@ -72,37 +72,23 @@ export class NgxD3timelineComponent
 
   @ViewChild('svgEl') svgEl: ElementRef<SVGElement>;
 
-  destroy$ = new Subject<boolean>();
-
   constructor(public timeline: NgxD3TimelineService) {}
 
   ngOnInit(): void {
-    this.observableToOutputMap.forEach(this.outputOnObservableEmit.bind(this));
+    this.initEventEmitters();
   }
 
   ngAfterViewInit(): void {
     this.timeline.setupZoom(this.svgEl);
   }
 
   ngOnDestroy(): void {
-    this.destroy$.next(true);
-  }
-
-  private get observableToOutputMap(): [Observable<any>, EventEmitter<any>][] {
-    return [
-      [this.timeline.hoveredActivity$, this.hovered],
-      [this.timeline.unhoveredActivity$, this.unhovered],
-      [this.timeline.activityDropped$, this.activityDropped]
-    ];
+    this.timeline.componentDestroyed();
```

Comment:
I had something like this and was getting an error when running the tests locally, as the service was destroyed before the component thus timeline was null. Might pay to check locally.

In any case it might be more appropriate to have the ngDestroy directly in the service? Its only usage is in the service so I think its redundant having it here and passing it through?

Reply:
Ran the tests locally didn't see any errors.

ngOnDeystroy is a component lifecycle hook. It doesn't get called by the service.